### ISSUE TITLE
arm-none-eabi GCC toolchain

### ISSUE THREAD

**Issue by DavidEGrayson** [msys2/MINGW-packages#842]
It would be nice to have a working arm-none-eabi GCC toolchain in MSYS2.  I'm making this issue to organize the discussion about that toolchain and figure out what its current state is.

@nanodude and @mingwandroid added scripts for mingw-w64-arm-none-eabi-binutils, mingw-w64-arm-none-eabi-newlib, mingw-w64-arm-none-eabi-gcc, mingw-w64-arm-none-eabi-gdb in pull request #338 back in 2014.  However, binaries are not available for those packages at http://repo.msys2.org/.  I tried to build those packages myself to figure out what is wrong.
- arm-none-eabi-binutils builds correctly on today's MSYS2.
- arm-none-eabi-newlib is supposed to be built next (it builds its own cross-compiler in the PKGBUILD's prepare() function), but that fails because cloog/cloog.h is not found.
- cloog has a PKGBUILD in this repository, but it does not build because a test fails:

```
Check file ./isl/jacobi-shared.cloog \c
(options -f 4 -l -1 -override -strides 1 -sh 1 ), \c
generating... \c
--- cloog_temp  2015-10-21 09:02:56.879971400 -0700
+++ ./isl/jacobi-shared.c       2013-10-11 00:27:03.000000000 -0700
@@ -3,7 +3,7 @@
   if ((16*floord(t0-1,16) >= -N+g1+t0+1) && (16*floord(g1+t0-3,16) >= -N+g1+t0+1) && (32*floord(t1-1,32) >= -N+g2+t1+1) && (32*floord(g2+t1-3,32) >= t1-32)) {
     for (c0=max(-16*floord(t0-1,16)+t0,-16*floord(g1+t0-3,16)+t0);c0<=min(32,N-g1-1);c0+=16) {
       for (c1=-32*floord(t1-1,32)+t1;c1<=min(32,N-g2-1);c1+=32) {
-        if (c1 >= 1) {
+        if ((c1 >= 1) && (c1 <= 32)) {
           S1(c0+g1-1,c1+g2-1);
         }
       }
FAIL: ./isl/jacobi-shared.c is not the same
```

So I think if we fix mingw-w64-cloog, that might allow the rest of the toolchain to build.  Please let me know if I'm on the wrong track.  Also, I intend to add cloog to the makedepends list of arm-none-eabi-gcc and arm-none-eabi-newlib if I get things working, so that it will be more obvious that it is needed.

**Comment by Alexpux** [comment]
GCC 5 drop clog support so we need to migrate ARM toolchains to gcc 5 too

**Comment by DavidEGrayson** [comment]
Thanks, @AlexPux.  That makes sense.  The gcc-arm-embedded project at launchpad has a [milestone for 5.0-2015-q4-major](https://launchpad.net/gcc-arm-embedded/+milestone/5.0-2015-q4-major) which makes it sound like they are going to release a GCC 5.0 toolchain before the end of the year.  I can probably just wait for that to happen.

**Comment by mingwandroid** [comment]
I'm quite active on the crosstool-ng project so I'll not spend time working on this package as it'd be at the expense of crosstool-ng. I do intend to improve that package in MSYS2 though (see how I avoided writing crosstool-ng a third time .. d'oh).

**Comment by DavidEGrayson** [comment]
Separately, there's got to be a better way to build newlib that doesn't involve rebuilding all of GCC each time in the prepare() function.  Hopefully we can figure that out when the time comes.  Maybe crosstool-ng will take care of that for us, @mingwandroid?

**Comment by mingwandroid** [comment]
crosstool-ng builds its C library only once, however newlib is pretty well deprecated these days. musl and uClibc-ng are the choices of a new generation.

**Comment by DaAwesomeP** [comment]
Any progress? This would be fantastic.

**Comment by DavidEGrayson** [comment]
I don't think so.  If you need to cross-compile something any time soon, you're probably better off looking for a Linux-based solution or downloading the [official GCC ARM toolchain](https://launchpad.net/gcc-arm-embedded/), which supports Windows.

**Comment by Trance-Paradox** [comment]
any news on the availability?
how can i get this pacakge

**Comment by DavidEGrayson** [comment]
I don't think MSYS2 has an ARM GCC toolchain yet, and they don't need one because it's so easy to just download the official one like I mentioned in my last comment.  You can add that toolchain to your MSYS2 PATH manually or with a simple shell script.  Doing it that way is nice because then you are in total control over which version you use, and the MSYS2 developers won't have to put effort maintaining it.

**Comment by Biswa96** [comment]
Does this answer your question https://github.com/msys2/MINGW-packages/issues/6844?
